captum/attr/_core/layer/layer_lrp.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+from ...._utils.common import _format_input
+from ...._utils.gradient import (
+    apply_gradient_requirements,
+    compute_gradients,
+    undo_gradient_requirements,
+)
+from ..._core.lrp import LRP
+from ..._utils.attribution import LayerAttribution
+
+
+class LayerLRP(LRP, LayerAttribution):
+    r"""
+    Layer-wise relevance propagation is based on a backward propagation
+    mechanism applied sequentially to all layers of the model. Here, the
+    model output score represents the initial relevance which is decomposed
+    into values for each neuron of the underlying layers. The decomposition
+    is defined by rules that are chosen for each layer, involving its weights
+    and activations. Details on the model can be found in the original paper
+    [https://doi.org/10.1371/journal.pone.0130140]. The implementation is
+    inspired by the tutorial of the same group
+    [https://doi.org/10.1016/j.dsp.2017.10.011] and the publication by
+    Ancona et al. [https://openreview.net/forum?id=Sy21R9JAW].
+    """
+
+    def __init__(self, model, layer):
+        """
+        Args:
+
+            model (callable): The forward function of the model or
+                        any modification of it. Custom rules for a given layer need to
+                        be defined as attribute
+                        `module.rule` and need to be of type PropagationRule.
+            layer (torch.nn.Module, None): Layer for which attributions are computed.
+                          The size and dimensionality of the attributions
+                          corresponds to the size and dimensionality of the layer's
+                          input or output depending on whether we attribute to the
+                          inputs or outputs of the layer. If value is None, the
+                          relevance for all layers is returned in attribution.
+        """
+        LayerAttribution.__init__(self, model, layer)
+        LRP.__init__(self, model)
+
+        self._check_rules()
```

Comment:
Isn't `_check_rules` already being called here ? 
https://github.com/pytorch/captum/blob/1838324879c0bcbccebad71f7e95b9a6eb85550d/captum/attr/_core/lrp.py#L50

Reply:
True, I removed the line.